hbase-zookeeper/src/main/java/org/apache/hadoop/hbase/zookeeper/ZKUtil.java

```diff
@@ -1760,6 +1760,10 @@ private static Op toZooKeeperOp(ZKWatcher zkw, ZKUtilOp op) throws UnsupportedOp
     }
   }
 
+  // Static boolean for warning about useMulti because we only want one warning per process
+  // instance.
+  private static boolean useMultiWarn = true;
```

Comment:
Non volatile and non atomic field which can be accessed multi threaded and we even change its value, always makes me nervous...

Please at least add a simple comment to say this is not a critical thing to have race and we log it multiple times, so we do not want to use AtomicBoolean here.

Reply:
I think simple comment is fine, because even if more than one thread enters the if block before reading the correctly reset value of boolean, we will still have quite limited occurrence of this logging, which is still much better than prior to this patch scenario.

Reply:
Makes sense. The thing that this static boolean controls, whether or not to log a warning, is trivial, so consequences for any multithreaded concern is also trivial. I will add a comment to this effect.